service/raft/raft_group_registry.cc

```diff
@@ -263,6 +263,23 @@ void raft_group_registry::init_rpc_verbs() {
             return rpc.execute_modify_config(from, std::move(add), std::move(del));
         });
     });
+
+    ser::raft_rpc_verbs::register_direct_fd_ping(&_ms,
+            [this] (const rpc::client_info&, raft::server_id dst) -> future<direct_fd_ping_reply> {
+        // XXX: update address map here as well?
+
+        const raft::server_id& my_id = get_my_raft_id();
+        if (my_id != dst) {
```

Comment:
What about other raft verbs, don't we need them to reject verbs arriving at the wrong destination?

Reply:
True, we probably should.

It's interesting, because I haven't encountered issues with other verbs during replace. At a glance, it seems that (when replacing with the same IP) the replacing node should be handling RPCs intended for the replaced node, as well as RPCs intended for itself, which could lead to some weird situations. Or maybe Raft is already implemented in a way that handles all of this correctly?

In any case, the other Raft verbs are already passing `dst_id`, we only need to start checking them.

I'll create a follow-up issue.

cc @kostja

Reply:
Opened: https://github.com/scylladb/scylladb/issues/12179